src/html.rs

```diff
@@ -450,4 +450,13 @@ impl<'s, I: Iterator<Item = Event<'s>>, W: std::fmt::Write> Writer<'s, I, W> {
         }
         self.out.write_str(s)
     }
+
+    fn write_escaped_char(&mut self, c: char) -> std::fmt::Result {
+        match c {
+            '<' => self.out.write_str("&lt;"),
+            '>' => self.out.write_str("&gt;"),
+            '&' => self.out.write_str("&amp;"),
+            c => self.out.write_char(c),
```

Comment:
We should actually also escape `"` here as `&quot;`. I didn't think it was needed earlier because it always turns into either a left or right quote in formatted text.

There is actually a test from the reference implementation for this that we should enable now:

```
---- suite::attributes::test_48546bb stdout ----
thread 'suite::attributes::test_48546bb' panicked at 'assertion failed: `(left == right)`
  left: `"<p><span key=\"\"#hi\">hi</span></p>"`,
 right: `"<p><span key=\"&quot;#hi\">hi</span></p>"`:
========================= INPUT ============================
hi{key="\"#hi"}
=================== ACTUAL vs EXPECTED =====================
<p><span key=""#hi">hi</span></p>    !=    <p><span key="&quot;#hi">hi</span></p>
============================================================
', tests/suite/attributes.rs:83:5
```

Actually, it should be escaped in the write_escape function also, currently we can get invalid HTML if we use `"` in urls, for example:
```
[text]("url)]
```
```html
<p><a href=""url"">text</a></p>
```

Reply:
Hmm, I was thinking so at first. But it should be possible to have unescaped quotes in normal text, as shown by the failing unit test.

I suppose we must have a flag of some sort, then. For normal text it should be unescaped, within attribute strings it should be escaped. We could have an `is_attribute` flag, and if it is set, we escape the quote. However, a bool is quite unreadable from the call side.

How about if we add a flag to `write_escape` and then we add two functions `write_text` and `write_attr` that simply calls the `write_escape`? And make sure the write_escape is never used directly.

Reply:
Fixed!